pkg/jwt/keywatch.go

```diff
@@ -0,0 +1,63 @@
+package jwt
+
+import (
+	"context"
+	"crypto/rsa"
+	"time"
+
+	"github.com/stackrox/rox/pkg/k8scfgwatch"
+)
+
+const (
+	watchInterval = 1 * time.Minute
+)
+
+var (
+	_ k8scfgwatch.Handler = (*handler)(nil) // compile time check that the handler implements the interface
+)
+
+type loadKeyFunc func(dir string) (*rsa.PrivateKey, error)
+type updateKeyFunc func(key *rsa.PrivateKey)
+
+// WatchKeyDir starts watching the directory containing JWT private key.
+func WatchKeyDir(dir string, loadKey loadKeyFunc, updateKey updateKeyFunc) {
```

Comment:
Since we constrain this to private keys, let's also make this obvious in the function name.